src/api/controller/api/publish.js

```diff
@@ -1,20 +1,40 @@
 /* eslint no-await-in-loop: off */
 import getDocumentTransformer from '../../../common/getDocumentTransformer';
 
+export const tranformAllDocument = async (count, findLimitFromSkip, insertMany, transformer) => {
+    let handled = 0;
+    while (handled < count) {
+        const dataset = await findLimitFromSkip(100, handled);
+        const transformedDataset = await Promise.all(dataset.map(transformer));
+        await insertMany(transformedDataset);
+        handled += dataset.length;
+    }
+};
+
 export const publishMiddleware = async (ctx) => {
     const count = await ctx.dataset.count({});
-    let handled = 0;
 
     const columns = await ctx.field.findAll();
 
-    const transformDocument = ctx.getDocumentTransformer({ env: 'node', dataset: ctx.dataset }, columns);
+    const uriIndex = columns.findIndex(col => col.name === 'uri');
```

Comment:
Can't you simply use `columns.find(col => col.name === 'uri')` here and pass the column found to `getDocumentTransformer` ?